src/validator.ts

```diff
@@ -2,30 +2,37 @@ import {
   FieldValidationFunctionSync,
   parseMessageWithCustomArgs,
 } from '@lemoncode/fonk';
+import { checkNIF, mapCustomArgsToMessageArgs } from './validator.business';
+import { CustomArgs } from './validator.model';
 
-// TODO: Add validator type
-const VALIDATOR_TYPE = '';
+const VALIDATOR_TYPE = 'IS_VALID_NIF';
 
-// TODO: Add default message
-let defaultMessage = '';
+let defaultMessage = 'The value must be a valid {{validTypes}}';
 export const setErrorMessage = message => (defaultMessage = message);
 
 const isDefined = value => value !== void 0 && value !== null && value !== '';
 
-export const validator: FieldValidationFunctionSync = fieldValidatorArgs => {
+const validateType = value => typeof value === 'string';
+
+const validate = (value: string, args: CustomArgs) =>
+  args.validTypes.indexOf(checkNIF(value)) >= 0;
```

Comment:
Why don't you use `some` method? I think it could be more readable if we have something like:

```
args.validTypes.some( type => isValidNIF(value, type))
```

What do you think?

Reply:
Done